### PR TITLE
Fix RuboCop warnings.

### PR DESCRIPTION
**Pull request by MikeMcQuaid**
Using the new rules in https://github.com/Homebrew/brew/pull/2679.

CC @ilovezfs @reitermarkus @GauthamGoli

### CODE REVIEW

**Comment by vszakats** [comment]
Minor nit: In `swimat.rb` `desc` "command-line Tool to help format Swift code", the word `tool` should probably become lowercase as well.

File: Formula/boost-python.rb [hunk 1]
```diff
@@ -2,8 +2,8 @@ class BoostPython < Formula
   desc "C++ library for C++/Python interoperability"
   homepage "https://www.boost.org/"
   url "https://downloads.sourceforge.net/project/boost/boost/1.63.0/boost_1_63_0.tar.bz2"
-  sha256 "beae2529f759f6b3bf3f4969a19c2e9d6f0c503edcb2de4a61d1428519fcb3b0"
   url "https://dl.bintray.com/boostorg/release/1.64.0/source/boost_1_64_0.tar.bz2"
+  sha256 "beae2529f759f6b3bf3f4969a19c2e9d6f0c503edcb2de4a61d1428519fcb3b0"
```
Comment (DomT4) [inline Formula/boost-python.rb]: There are two `sha256`s here.

File: Formula/pachi.rb [hunk 2]
```diff
@@ -33,13 +33,9 @@ def install
     system "make"
     bin.install "pachi"
 
-    if build.with? "patterns"
-      share.install resource("patterns")
-    end
+    share.install resource("patterns") if build.with? "patterns"
```
Comment (DomT4) [inline Formula/pachi.rb]: Not sure this shouldn't be `pkgshare` really, although that's not changed by your rearranging.

File: Formula/paps.rb [hunk 3]
```diff
@@ -53,7 +53,7 @@ def install
 --- a/src/libpaps.c
 +++ b/src/libpaps.c
 @@ -25,8 +25,10 @@
- 
+
```
Comment (DomT4) [inline Formula/paps.rb]: Patch whitespace got tweaked here.

File: Formula/sleuthkit.rb [hunk 4]
```diff
@@ -11,19 +11,19 @@ class Sleuthkit < Formula
...
 
-  conflicts_with "irods", :because => "both install `ils`"
-
   option "with-jni", "Build Sleuthkit with JNI bindings"
   option "with-debug", "Build debug version"
 
+  depends_on "afflib" => :optional
+  depends_on "libewf" => :optional
+
+  conflicts_with "irods", :because => "both install `ils`"
```
Comment (DomT4) [inline Formula/sleuthkit.rb]: This hasn't quite ended up in the right place. I presume `rubocop` tripped over there being nested `if build.with?` deps below.

File: Formula/stone.rb [hunk 5]
```diff
@@ -15,7 +15,7 @@ class Stone < Formula
   option "with-ssl", "SSL support"
 
   def install
-    target = (build.with? "ssl") ? "macosx-ssl" : "macosx"
+    target = build.with?("ssl") ? "macosx-ssl" : "macosx"
```
Comment (DomT4) [inline Formula/stone.rb]: Unrelated but I'd probably bet money that this is trying to use the system OpenSSL.

File: Formula/boost-python.rb (hunk 1 above)
Comment (GauthamGoli) [inline Formula/boost-python.rb]: This and another audit error of `boost-python.rb` is being fixed in #13375

File: Formula/tenyr.rb [hunk 6]
```diff
@@ -30,18 +30,14 @@ def install
     # JIT build is not available until we can pull in AsmJit somehow
     # HEAD version can build with JIT enabled, using git submodule
     # Right now there is no way for `build.with?("jit")` to be true
-    if build.without?("jit")
-      args << "JIT=0"
-    end
+    args << "JIT=0" if build.without?("jit")
```
Comment (DomT4) [inline Formula/tenyr.rb]: The `()` are redundant here, although that has never been consistently enforced on formulae.

File: Formula/sleuthkit.rb (hunk 4 above)
Comment (GauthamGoli) [inline Formula/sleuthkit.rb]: Even this is being fixed in #13375

File: Formula/cdogs-sdl.rb [hunk 7]
```diff
@@ -24,7 +24,7 @@ def install
     system "make"
     prefix.install "src/cdogs-sdl.app"
     bin.write_exec_script "#{prefix}/cdogs-sdl.app/Contents/MacOS/cdogs-sdl"
-    pkgshare.install ["data", "dogfights", "graphics", "missions", "music", "sounds"]
+    pkgshare.install %w[data dogfights graphics missions music sounds]
```
Comment (reitermarkus) [inline Formula/cdogs-sdl.rb]: @MikeMcQuaid, I'm thinking the same here as with symbol arrays.

**Review by reitermarkus — DISMISSED** [review]

File: Formula/cdogs-sdl.rb (hunk 7 above)
Comment (ilovezfs) [inline Formula/cdogs-sdl.rb]: This rule seems yuck to me.

File: Formula/coreutils.rb [hunk 8]
```diff
@@ -69,9 +69,7 @@ def install
     # https://github.com/Homebrew/homebrew/issues/44993
     # This is thought to be an el_capitan bug:
     # https://lists.gnu.org/archive/html/bug-tar/2015-10/msg00017.html
-    if MacOS.version == :el_capitan
-      ENV["gl_cv_func_getcwd_abort_bug"] = "no"
-    end
+    ENV["gl_cv_func_getcwd_abort_bug"] = "no" if MacOS.version == :el_capitan
```
Comment (ilovezfs) [inline Formula/coreutils.rb]: This rule will end up causing things to go passed the eightieth column.

File: Formula/aubio.rb [hunk 9]
```diff
@@ -25,9 +25,7 @@ class Aubio < Formula
   depends_on "fftw" => :optional
   depends_on "jack" => :optional
 
-  if build.with? "python"
-    depends_on "numpy" => :python
-  end
+  depends_on "numpy" => :python if build.with? "python"
```
Comment (ilovezfs) [inline Formula/aubio.rb]: This was to avoid audit failure.

File: Formula/fuseki.rb [hunk 10]
```diff
@@ -38,8 +38,6 @@ def install
     prefix.install "Data"
   end
 
-  plist_options :manual => "fuseki start"
```
Comment (ilovezfs) [inline Formula/fuseki.rb]: This will conflict GauthamGoli's components PR.

File: Formula/gdal.rb [hunk 11]
```diff
@@ -83,8 +84,6 @@ class Gdal < Formula
     depends_on "ant" => :build
     depends_on "swig" => :build
   end
-
-  option "without-python", "Build without python2 support"
```
Comment (ilovezfs) [inline Formula/gdal.rb]: This will also conflict GauthamGoli's components PR.

**Comment by ilovezfs** [comment]
I very much not a fan of the %w thing or the forced modifier if.

**Comment by ilovezfs** [comment]
https://github.com/Homebrew/homebrew-core/pull/13375 should be merged before this PR.

File: Formula/paps.rb (hunk 3 above)
Comment (ilovezfs) [inline Formula/paps.rb]: Was this caused by a rubocop autofix? If so, that is a problem with the autofix not the formula.

File: Formula/hashcat.rb [hunk 12]
```diff
@@ -60,7 +60,7 @@ def install
     # Test 1 (dictionary attack, -a 0):
     #
 
-    hash = File.open(hash_file, "rb").read.strip
+    hash = File.open(hash_file, "rb") { |f| f.read.strip }
```
Comment (ilovezfs) [inline Formula/hashcat.rb]: Why is the replacement better? That seems worse.

**Comment by ilovezfs** [comment]
I've merged #13375 so those changes should be removed here.

File: Formula/kube-aws.rb [hunk 13]
```diff
@@ -44,7 +44,7 @@ def install
     system "#{bin}/kube-aws"
     cluster = { "clusterName"=>"test-cluster",
                 "apiEndpoints"=>[{ "name"=>"default", "dnsName"=>"dns",
-                "loadBalancer"=>{ "createRecordSet"=>false } }],
+                                   "loadBalancer"=>{ "createRecordSet"=>false } }],
```
Comment (ilovezfs) [inline Formula/kube-aws.rb]: Not sure what's going on here with this change. The alignment looks like it was more correct before than after.

File: Formula/swimat.rb [hunk 14]
```diff
@@ -1,5 +1,5 @@
 class Swimat < Formula
-  desc "Command Line Tool to help format Swift code"
+  desc "command-line Tool to help format Swift code"
```
Comment (ilovezfs) [inline Formula/swimat.rb]: This results in incorrect capitalization of the description.

File: Formula/paps.rb (hunk 3 above)
Comment (MikeMcQuaid) [inline Formula/paps.rb]: It was caused by Mike sucking.

File: Formula/cdogs-sdl.rb (hunk 7 above)
Comment (MikeMcQuaid) [inline Formula/cdogs-sdl.rb]: Note: this is only on longer string arrays.
@reitermarkus Why in this case? Your grepping argument doesn't stand up here so much.

File: Formula/coreutils.rb (hunk 8 above)
Comment (MikeMcQuaid) [inline Formula/coreutils.rb]: @ilovezfs Formerly, yes, but RuboCop now handles it and has a specific line length that it applies it to. Currently this is 80 but can be shortened if desired.

File: Formula/aubio.rb (hunk 9 above)
Comment (MikeMcQuaid) [inline Formula/aubio.rb]: That's cool, I think I've now fixed the whining so we'll see if 💥

File: Formula/hashcat.rb (hunk 12 above)
Comment (MikeMcQuaid) [inline Formula/hashcat.rb]: Previously it was not closing the file descriptor. Not a terrible problem here as it'll be closed when the entire process exits but if you do that in a loop: 💥. It's bad practise, regardless.

File: Formula/kube-aws.rb (hunk 13 above)
Comment (MikeMcQuaid) [inline Formula/kube-aws.rb]: Agreed but I think it was also weird before. Will tweak manually.

File: Formula/swimat.rb (hunk 14 above)
Comment (MikeMcQuaid) [inline Formula/swimat.rb]: This was Mike fail again.

File: Formula/hashcat.rb (hunk 12 above)
Comment (ilovezfs) [inline Formula/hashcat.rb]: That's a good reason to change it.

File: Formula/aubio.rb (hunk 9 above)
Comment (ilovezfs) [inline Formula/aubio.rb]: Still whines:
```
  * Dependency numpy should not use option python
```

File: Formula/coreutils.rb (hunk 8 above)
Comment (ilovezfs) [inline Formula/coreutils.rb]: This reads like gibberish to me, though:
```
args << "--enable-nonfree" if %w[fdk-aac openssl].any? { |f| build.with? f }
```
So I'm not sure enforcing this as an automated rule is a great idea.

File: Formula/pachi.rb (hunk 2 above)
Comment (MikeMcQuaid) [inline Formula/pachi.rb]: @DomT4 Agreed but not going to pull bottles for this so avoiding changing any behaviour here. Could you make a PR?

File: Formula/stone.rb (hunk 5 above)
Comment (MikeMcQuaid) [inline Formula/stone.rb]: @DomT4 Agreed but not going to pull bottles for this so avoiding changing any behaviour here. Could you make a PR?

File: Formula/cdogs-sdl.rb (hunk 7 above)
Comment (MikeMcQuaid) [inline Formula/cdogs-sdl.rb]: Spoke to @ilovezfs in Slack and he seemed to not mind this too much given it's only for string arrays that are >=4 long.

File: Formula/coreutils.rb (hunk 8 above)
Comment (MikeMcQuaid) [inline Formula/coreutils.rb]: Yeh, fixing that gibberish here because someone tried to be too clever previously.

File: Formula/aubio.rb (hunk 9 above)
Comment (MikeMcQuaid) [inline Formula/aubio.rb]: Cool, will change back.

**Review by ilovezfs — APPROVED** [review]

File: Formula/stone.rb (hunk 5 above)
Comment (DomT4) [inline Formula/stone.rb]: Wouldn't need to pull bottles here given it's optional, but happy to file a PR, aye.

File: Formula/pachi.rb (hunk 2 above)
Comment (DomT4) [inline Formula/pachi.rb]: Aye.

**Pull request merged**